sqlmesh/dbt/context.py

```diff
@@ -118,6 +127,10 @@ def _var(name: str, default: t.Optional[t.Any] = None) -> t.Any:
 
         self.variables = rendered_variables
 
+    def add_macros(self, macros: t.Dict[str, MacroInfo], package: str) -> None:
```

Comment:
why is this touched, cc @crericha

Reply:
Why - jinja environment was cached and wrong when created prior to adding macros. This triggers recreation of the jinja environment. 

How'd it get here - It was part of the dbt-default-database branch merged into this branch that contained fixes made for the recent dbt project we worked on. It could be split out into its own PR, but I'm not sure its worth the effort to do so. Up to you.

Reply:
leave it